awswrangler/s3/_write_dataset.py

```diff
@@ -18,6 +19,26 @@
 _logger: logging.Logger = logging.getLogger(__name__)
 
 
+def _load_mode_and_filename_prefix(*, mode: str | None, filename_prefix: str | None = None) -> tuple[str, str]:
+    if mode is None:
+        mode = "append"
+
+    if mode == "overwrite_files":
+        # In `overwrite_files` mode, we need to create deterministic
+        # filenames to ensure that the same files are always overwritten:
+        if filename_prefix is None:
+            filename_prefix = "data"
+        random_filename_suffix = ""
+        mode = "append"
```

Comment:
It's a bit confusing why we are changing the mode to `append` when it's originally passed as `overwrite_files`. It could cause unforeseen errors in the future and make tracing/debugging difficult